pkg/api/alerting.go

```diff
@@ -507,91 +507,107 @@ func (hs *HTTPServer) NotificationTest(c *models.ReqContext) response.Response {
 }
 
 // POST /api/alerts/:alertId/pause
-func (hs *HTTPServer) PauseAlert(c *models.ReqContext) response.Response {
-	dto := dtos.PauseAlertCommand{}
-	if err := web.Bind(c.Req, &dto); err != nil {
-		return response.Error(http.StatusBadRequest, "bad request data", err)
-	}
-	alertID, err := strconv.ParseInt(web.Params(c.Req)[":alertId"], 10, 64)
-	if err != nil {
-		return response.Error(http.StatusBadRequest, "alertId is invalid", err)
-	}
-	result := make(map[string]interface{})
-	result["alertId"] = alertID
-
-	query := models.GetAlertByIdQuery{Id: alertID}
-	if err := hs.SQLStore.GetAlertById(c.Req.Context(), &query); err != nil {
-		return response.Error(500, "Get Alert failed", err)
+func (hs *HTTPServer) PauseAlert(legacyAlertingEnabled *bool) func(c *models.ReqContext) response.Response {
+	if legacyAlertingEnabled == nil || !*legacyAlertingEnabled {
```

Comment:
Based on the following:
https://github.com/grafana/grafana/blob/62b0a8bae6abcbd75326d3310abebcf6a78eac78/pkg/setting/setting.go#L1427-L1431

https://github.com/grafana/grafana/blob/62b0a8bae6abcbd75326d3310abebcf6a78eac78/pkg/setting/setting_unified_alerting.go#L119-L122

I think that the only case `AlertingEnabled` is `nil` is when the `alerting.enabled` cannot be parsed and in that case the new alerting is enabled.
So, I think the check should become:
```suggestion
	if legacyAlertingEnabled != nil || !*legacyAlertingEnabled {
```